packages/flutter_localizations/test/text_test.dart

```diff
@@ -0,0 +1,183 @@
+    final Key targetKey = UniqueKey();
+    await tester.pumpWidget(
+      MaterialApp(
+        routes: <String, WidgetBuilder>{
+          '/next': (BuildContext context) {
+            return const Text('Next');
+          },
+        },
+        localizationsDelegates: const <LocalizationsDelegate<dynamic>>[
+          GlobalMaterialLocalizations.delegate,
+        ],
+        supportedLocales: const <Locale>[
+          Locale('en', 'US'),
+          Locale('es', 'ES'),
+          Locale('zh', 'CN'),
+        ],
+        locale: const Locale('zh', 'CN'),
+        home: Material(
+          child: Center(
+            child: Builder(
+              key: targetKey,
+              builder: (BuildContext context) {
+                return PopupMenuButton<int>(
+                  onSelected: (int value) {
+                    Navigator.pushNamed(context, '/next');
+                  },
+                  itemBuilder: (BuildContext context) {
+                    return <PopupMenuItem<int>>[
+                      PopupMenuItem<int>(
+                        value: 1,
+                        child: Text(
+                          'hello, world',
+                          style: TextStyle(color: Colors.blue),
+                        ),
+                      ),
+                      PopupMenuItem<int>(
+                        value: 2,
+                        child: Text(
+                          '你好，世界',
+                          style: TextStyle(color: Colors.blue),
+                        ),
+                      ),
+                    ];
+                  },
+                );
+              },
+            ),
+          ),
+        ),
+      ),
+    );
+
+    await tester.tap(find.byKey(targetKey));
+    await tester.pumpAndSettle();
+
+    expect(find.text('hello, world'), findsOneWidget);
+    expect(find.text('你好，世界'), findsOneWidget);
+
+    Offset topLeft = tester.getTopLeft(find.text('hello, world'));
+    Offset topRight = tester.getTopRight(find.text('hello, world'));
+    Offset bottomLeft = tester.getBottomLeft(find.text('hello, world'));
+    Offset bottomRight = tester.getBottomRight(find.text('hello, world'));
+
+    expect(topLeft, const Offset(392.0, 298.3999996185303));
+    expect(topRight, const Offset(596.0, 298.3999996185303));
+    expect(bottomLeft, const Offset(392.0, 315.3999996185303));
+    expect(bottomRight, const Offset(596.0, 315.3999996185303));
+
+    topLeft = tester.getTopLeft(find.text('你好，世界'));
+    topRight = tester.getTopRight(find.text('你好，世界'));
+    bottomLeft = tester.getBottomLeft(find.text('你好，世界'));
+    bottomRight = tester.getBottomRight(find.text('你好，世界'));
+
+    expect(topLeft, const Offset(392.0, 346.3999996185303));
+    expect(topRight, const Offset(477.0, 346.3999996185303));
+    expect(bottomLeft, const Offset(392.0, 363.3999996185303));
+    expect(bottomRight, const Offset(477.0, 363.3999996185303));
+  }, skip: !isLinux);
```

Comment:
Any way we can make the tests pass on all platforms?

Reply:
Not easily, since each platform reads the fonts differently depending on if it is using Coretext or OpenType.

Reply:
This is also the reason why, we only run libtxt unit tests on linux.

Reply:
Maybe add at least a comment describing that?